lib/elixir/test/elixir/kernel/cli_test.exs

```diff
@@ -30,6 +30,23 @@ defmodule Kernel.CLITest do
            end)
   end
 
+  test "--version smoke test" do
+    output = elixir('--version')
+    assert output =~ ~r/Erlang\/OTP [0-9]+ \[.+]/
+    assert output =~ ~r/Elixir [1-9]\.[0-9]+\.[0-9]+.*\(compiled with Erlang\/OTP [0-9]+\)/
+  end
+
+  test "--short-version smoke test" do
+    output = elixir('--short-version')
+    assert output =~ ~r/^[1-9]\.[0-9]+\.[0-9]+(?:-dev)?\n$/m
```

Comment:
What do you think about checking that both `System.version/0` and `--short-version` match?

```suggestion
    assert output == System.version() <> "\n"
```